File: scripts/generate_students.py
import pandas as pd
import random

zip_codes = ["75069", "75070", "75071", "75072"]

students = []

for i in range(1, 101):
    students.append({
        "student_id": i,
        "zip_code": random.choice(zip_codes)
    })

df = pd.DataFrame(students)

df.to_csv("data/students.csv", index=False)

print("Generated 100 students!")

File: scripts/generateStudents.js
const fs = require("fs");


// Heritage High School boundary box
const minLat = 33.175;
const maxLat = 33.207;

const minLng = -96.790;
const maxLng = -96.728;



function pointInside(lat,lng){


return (

lat >= minLat &&
lat <= maxLat &&
lng >= minLng &&
lng <= maxLng

);


}



const students=[];


let attempts=0;



while(
students.length < 200 &&
attempts < 50000
){


attempts++;


const lat =
minLat +
Math.random() *
(maxLat-minLat);



const lng =
minLng +
Math.random() *
(maxLng-minLng);



if(pointInside(lat,lng)){


students.push({

student_id:
students.length+1,


lat:
Number(lat.toFixed(6)),


lng:
Number(lng.toFixed(6)),


zip_code:
"75035"


});


}


}



fs.writeFileSync(

"public/students.json",

JSON.stringify(
students,
null,
2
)

);



console.log(
"Generated",
students.length,
"students"
);